ugvc/mrd/srsnv_training_utils.py

```diff
@@ -165,10 +165,14 @@ def prepare_featuremap_for_model(
             featuremap_entry_number += 1
     if featuremap_entry_number < total_size:
         logger.warning(
-            f"featuremap_entry_number={featuremap_entry_number} > training_set_size={train_set_size}"
+            "Requested training and test set size cannot be met - insufficient data"
+            f"featuremap_entry_number={featuremap_entry_number} < training_set_size={train_set_size}"
             f"+ test_set_size={test_set_size if test_set_size else 0}"
-            "\nbehavior is undefined",
-        )  # TODO in this case we should make sure we still have a test set (and behavior should be defined)
+        )
+        if test_set_size is not None:
+            train_set_size = np.floor(featuremap_entry_number * (train_set_size / total_size))
```

Comment:
the total_size is the total size of the featuremap?
So if test-size is None we use all remaining rows as test?

Reply:
total_size = train_set_size + test_set_size

Removing the option to have None

Reply:
Wait. 
if featuremap_entry_number is the size of the featuremap (in the region)
and total_size is train+test 
this calculation seems weird isn't it?